tests/unit/amo/components/TestEditableCollectionAddon.js

```diff
@@ -76,4 +83,164 @@ describe(__filename, () => {
+      expect(root.find('.EditableCollectionAddon-notes-read-only'))
+        .toHaveLength(0);
+      expect(notesForm).toHaveProp('microButtons', true);
+      expect(notesForm).toHaveProp('onDelete', root.instance().onDeleteNote);
+      expect(notesForm).toHaveProp('onDismiss', root.instance().onDismissNoteForm);
+      expect(notesForm).toHaveProp('onSubmit', root.instance().onSaveNote);
+      expect(notesForm).toHaveProp('placeholder', 'Add a comment about this add-on.');
+      expect(notesForm).toHaveProp('submitButtonText', 'Save');
+      expect(notesForm).toHaveProp('text', undefined);
+    });
+
+    it('shows a populated notes form when the edit button icon is clicked', () => {
+      const notes = 'Some notes.';
+      const addon = {
+        ...fakeAddon,
+        notes,
+      };
+
+      const root = render({ addon });
+      expect(root.find('.EditableCollectionAddon-notes-form')).toHaveLength(0);
+      const editButton = root.find('.EditableCollectionAddon-notes-edit-button');
+      const clickEvent = createFakeEvent();
+      editButton.simulate('click', clickEvent);
+      const notesForm = root.find('.EditableCollectionAddon-notes-form');
+      expect(notesForm).toHaveLength(1);
+      // The read-only portion should not be shown.
+      expect(root.find('.EditableCollectionAddon-notes-read-only'))
+        .toHaveLength(0);
+      expect(notesForm).toHaveProp('text', notes);
+    });
+
+    it('hides the notes form when the cancel button is clicked on the DismissibleTextForm', () => {
+      const notes = 'Some notes.';
+      const addon = {
+        ...fakeAddon,
+        notes,
+      };
+
+      let root = render({ addon });
+      expect(root.find('.EditableCollectionAddon-notes-form')).toHaveLength(0);
+      const editButton = root.find('.EditableCollectionAddon-notes-edit-button');
+      const clickEvent = createFakeEvent();
+      editButton.simulate('click', clickEvent);
+      let notesForm = root.find('.EditableCollectionAddon-notes-form');
+      expect(notesForm).toHaveLength(1);
+
+      // This simulates the user clicking the "Cancel" button on the
+      // DismissibleTextForm.
+      const onDismissNoteForm = notesForm.props().onDismiss;
+      onDismissNoteForm();
+
+      root = render({ addon });
+      notesForm = root.find('.EditableCollectionAddon-notes-form');
+      expect(notesForm).toHaveLength(0);
+    });
+
+    it('calls deleteNote when the delete button is clicked on the DismissibleTextForm', () => {
+      const notes = 'Some notes.';
+      const addon = {
+        ...fakeAddon,
+        notes,
+      };
+      const deleteNote = sinon.spy();
+      const errorHandler = createStubErrorHandler();
+
+      const root = render({ addon, deleteNote, errorHandler });
+
+      expect(root.find('.EditableCollectionAddon-notes-form')).toHaveLength(0);
+      const editButton = root.find('.EditableCollectionAddon-notes-edit-button');
+      const clickEvent = createFakeEvent();
```

Comment:
I don't think we are using this variable more than once, so I'd suggest to not create it.